src/Microsoft.Azure.ServiceBus/SessionReceivePump.cs

```diff
@@ -323,7 +314,7 @@ void RaiseExceptionRecieved(Exception e, string action)
                     // Lets not bother user with this exception.
                     if (!(exception is TaskCanceledException))
                     {
-                        this.sessionHandlerOptions.RaiseExceptionReceived(new ExceptionReceivedEventArgs(exception, ExceptionReceivedEventArgsAction.RenewLock));
+                        this.sessionHandlerOptions.RaiseExceptionReceived(new ExceptionReceivedEventArgs(exception, ExceptionReceivedEventArgsAction.RenewLock, namespaceName));
```

Comment:
`CloseSessionIfNeededAsync()` invokes `RaiseExceptionReceived()` which raises `ExceptionReceivedEventArgs` in a centralized manner, appending to the passed in information the namespace name @nemakam . Or am I missing something else?